### PR TITLE
[WFLY-20040] Cannot check null values against collections which do no…

### PR DESCRIPTION
**Pull request by jamezp**
…t allow null values. Null check the value before checking the collection.

https://issues.redhat.com/browse/WFLY-20040

### CODE REVIEW

File: jaxrs/src/main/java/org/jboss/as/jaxrs/deployment/JaxrsScanningProcessor.java [hunk 1]
```diff
@@ -197,13 +197,13 @@ protected boolean hasBootClasses(JBossWebMetaData webdata) {
...
                 String servletClass = servlet.getServletClass();
-                if (BOOT_CLASSES.contains(servletClass))
+                if (servletClass != null && BOOT_CLASSES.contains(servletClass))
                     return true;
             }
         }
         if (webdata.getFilters() != null) {
             for (FilterMetaData filter : webdata.getFilters()) {
-                if (BOOT_CLASSES.contains(filter.getFilterClass()))
+                if (filter != null && BOOT_CLASSES.contains(filter.getFilterClass()))
```
Comment (scottmarlow) [inline jaxrs/src/main/java/org/jboss/as/jaxrs/deployment/JaxrsScanningProcessor.java:206]: Should this also check if filter.getFilterClass()) returns null?

Comment (jamezp) [reply jaxrs/src/main/java/org/jboss/as/jaxrs/deployment/JaxrsScanningProcessor.java:206]: Oops, yes. Nice catch.

[the remark this answers is not in the document]
Comment (jamezp) [reply jaxrs/src/main/java/org/jboss/as/jaxrs/deployment/JaxrsScanningProcessor.java:206]: Oops, yes. Nice catch :)

**Review by bstansberry — APPROVED** [review]

**Review by scottmarlow — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @jamezp and @scottmarlow

**Pull request merged**